compiler/rustc_borrowck/src/borrowck_errors.rs

```diff
@@ -29,17 +30,18 @@ impl<'cx, 'tcx> crate::MirBorrowckCtxt<'cx, 'tcx> {
         borrow_span: Span,
         borrow_desc: &str,
     ) -> DiagnosticBuilder<'tcx, ErrorGuaranteed> {
-        let mut err = struct_span_err!(
-            self,
-            span,
-            E0503,
-            "cannot use {} because it was mutably borrowed",
-            desc,
-        );
-
-        err.span_label(borrow_span, format!("{} is borrowed here", borrow_desc));
-        err.span_label(span, format!("use of borrowed {}", borrow_desc));
-        err
+        create_err(CannotUseWhenMutablyBorrowed { span, desc })
```

Comment:
@davidtwco Not sure how to use the create_err here after creating the struct

Reply:
Use `self.infcx.tcx.sess.create_err`, `create_err` isn't a free-standing function :)